### PR TITLE
Enable responsive syntax for all of `<Box>`'s style props

### PR DESCRIPTION
**Pull request by jesstelford**
Depends on https://github.com/Shopify/polaris/pull/10489

This PR:

- Enables responsive syntax for _all_ of `<Box>`'s style props
- Sets us up to remove usages of `useBreakpoints` internally (see https://github.com/Shopify/polaris/pull/10891)

### CODE REVIEW

File: polaris-react/src/components/Box/Box.scss [hunk 1]
```diff
@@ -8,110 +8,41 @@
...
+  @include responsive-props('box', 'inset-inline-end', 'inset-inline-end');
+  @include responsive-props('box', 'inset-inline-start', 'inset-inline-start');
+  @include responsive-props('box', 'max-width', 'max-width');
+  @include responsive-props('box', 'min-height', 'min-height');
+  @include responsive-props('box', 'min-width', 'min-width');
+  @include responsive-props('box', 'outline-color', 'outline-color');
+  @include responsive-props('box', 'outline-style', 'outline-style');
+  @include responsive-props('box', 'outline-width', 'outline-width');
+  @include responsive-props('box', 'overflow-x', 'overflow-x');
+  @include responsive-props('box', 'overflow-y', 'overflow-y');
```
Comment (aaronccasanova) [inline polaris-react/src/components/Box/Box.scss:36]: I'm all for this update! Might be a good opportunity to revisit my comment on the original implementation. Thoughts?

https://github.com/Shopify/polaris/pull/7697#discussion_r1023284163

Comment (jesstelford) [reply polaris-react/src/components/Box/Box.scss:36]: Oooh, yes, love it! Let's get that into a fast-follow PR 👍 

There's also an opportunity to optimize it even further by not even outputting a media query if that value is just going to fallback to an earlier breakpoint's value anyway.

**Review by laurkim — COMMENTED** [review]
Looks great overall! 💯 
Noticing some regressions with text when running locally though where the `color` isn't being set properly. That might be what's causing the large amount of baseline changes that need to be accepted for Storybook.

**Comment by jesstelford** [comment]
@laurkim Good catch! I realised it was to do with `initial` vs `inherit`, and fixed that issue in https://github.com/Shopify/polaris/pull/10489#issuecomment-1751642077 (along with an explanation of the fix), then rebased this branch on top so it _inherits_ that fix 👍 

Will watch to see if Chromatic shows fewer changes now 🤞 

_EDIT_: Yep, that worked 🎉 The "19 changes" showing at the moment are from the base branch. See [more details here](https://github.com/Shopify/polaris/pull/10489#issuecomment-1751841523).

File: polaris-react/src/utilities/css.ts [hunk 2]
```diff
@@ -170,17 +174,57 @@ export function getResponsiveValue<T extends string | number = string>(
   };
 }
 
-export function mapResponsivePropValues<Input, Output>(
-  responsiveProp: ResponsiveProp<Input> | undefined,
-  fn: (value?: Input) => Output | undefined,
-): ResponsiveProp<Output> | undefined {
+/*
```
Comment (laurkim) [inline polaris-react/src/utilities/css.ts:177]: Do we need the commented out code on lines 177-182 and 219-227?

**Review by laurkim — APPROVED** [review]

**Pull request closed without merging**